src/node/desktop/src/core/user.ts

```diff
@@ -19,21 +19,28 @@ import { getenv } from './environment';
 import { FilePath } from './file-path';
 
 import desktop from '../native/desktop.node';
+import { removeTrailingSlashes } from '../main/utils';
 
 export function userHomePath(): FilePath {
-  const user = getenv('R_USER');
-  if (checkPath(user)) return new FilePath(user);
-  const home = getenv('HOME');
-  if (checkPath(home)) return new FilePath(home);
+  const pathsToCheck = [() => getenv('HOME')];
   if (process.platform === 'win32') {
-    const currentHome = desktop.currentCSIDLPersonalHomePath();
-    if (checkPath(currentHome)) return new FilePath(currentHome);
-    const defaultHome = desktop.defaultCSIDLPersonalHomePath();
-    if (checkPath(defaultHome)) return new FilePath(defaultHome);
+    pathsToCheck.push(
+      () => getenv('R_USER'),
```

Comment:
This check should become before `HOME`, as R will check that before `R_USER`. See https://github.com/wch/r-source/blob/9d27b1c9ad917f0c8fbe8f6476927a1bd8ed9d9d/src/gnuwin32/sys-win32.c#L76-L121 for how this is handled in R on Windows (we try to stay true to pre-existing behaviors in R whenever possible)

Reply:
thanks for pointing this out!